src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,113 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+import { isValidHttpUrl } from '../../Helpers';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImbdUrl] = useState('');
+  const [imdbId, setImbdId] = useState('');
+
+  const isRequired = title
+    && isValidHttpUrl(imgUrl)
+    && isValidHttpUrl(imdbUrl)
+    && imdbId;
+
+  const clearForm = () => {
+    setCount(current => current + 1);
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImbdId('');
+    setImbdUrl('');
+  };
+
+  const newMovie = {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onAdd(newMovie);
+    clearForm();
+  };
+
+  const handleTitle = (value: string) => {
+    setTitle(value);
+  };
+
+  const handleDescription = (value: string) => {
+    setDescription(value);
+  };
+
+  const handleImgUrl = (value: string) => {
+    setImgUrl(value);
+  };
+
+  const handleImbdUrl = (value: string) => {
+    setImbdUrl(value);
+  };
+
+  const handleImbdId = (value: string) => {
+    setImbdId(value);
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={handleSubmit}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={(value) => handleTitle(value)}
```

Comment:
Just keep in mind that a components like `TextField` calls your function and provides all the props, so we don't need to create an extra function.
```suggestion 
        onChange={handleTitle}
```

Reply:
fixed 👌